vulnerabilities/helpers.py

```diff
@@ -100,9 +118,41 @@ def requests_with_5xx_retry(max_retries=5, backoff_factor=0.5):
     return session
 
 
-def contains_alpha(string):
-    """
-    Return True if the input 'string' contains any alphabet
-    """
+def nearest_patched_package(vulnerable_packages, resolved_packages):
+    # This class is used to  get around bisect module's lack of supplying custom
+    # compartor. Get rid of this once we use python 3.10 which supports this.
+    # See https://github.com/python/cpython/pull/20556
+    class PackageURLWithVersionComparator:
```

Comment:
Are you missing a @total_ordering there?

Also you should mention this is only to compare packages of the same type OR you could implement comparison and sorting across types.

Reply:
You may want to check that the type/ns/name are the same  too